core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -362,4 +362,8 @@ private TableProperties() {}
 
   public static final String UPSERT_ENABLED = "write.upsert.enabled";
   public static final boolean UPSERT_ENABLED_DEFAULT = false;
+  // Controls whether the set distribution mode has to be followed or not.
+  public static final String STRICT_TABLE_DISTRIBUTION_AND_ORDERING =
+      "strict-table-distribution-and-ordering";
+  public static final String STRICT_TABLE_DISTRIBUTION_AND_ORDERING_DEFAULT = "true";
```

Comment:
New table level property introduced to decide whether we want strict distribution mode specified in properties. As Rebalance supports only hash partitioning.

Reply:
minor suggestion here, since this is spark specific. maybe call this "write.spark.strict-table-distribution-ordering" Comment also should reflect that this is a spark property

Reply:
I believe this is only going to be for Spark 3.3 as well right? Should probably document that since we won't be doing a patch in 3.4 ?

Reply:
The doc which is currently in the planning section should probably be here instead

```
       // if strict distribution mode is not enabled, then we fallback to spark AQE
          // to determine the number of partitions by colaesceing and un-skewing partitions
          // Also to note, Rebalance is only supported for hash distribution mode till spark 3.3
          // By default the strictDistributionMode is set to true, to not disrupt regular
          // plan of RepartitionByExpression
          ```

Reply:
I don't think it is a good idea to add a new public-facing table property that is already applicable only to older versions of Spark. Can we add a SQL property instead in our 3.3 module? Also, I would be okay skipping it and just assuming the distribution has to be strict. Then Spark will coalesce small files but won't split large ones.

Also, the ordering is always required. It is just the distribution that can be strict or not. It should be reflected in the name of the property.